core/src/test/java/org/apache/struts2/views/jsp/ui/ScriptTagTest.java

```diff
@@ -0,0 +1,42 @@
+package org.apache.struts2.views.jsp.ui;
+
+import com.opensymphony.xwork2.ActionContext;
+import org.apache.struts2.util.TokenHelper;
+import org.apache.struts2.views.jsp.AbstractUITagTest;
+
+import javax.servlet.jsp.JspException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ScriptTagTest extends AbstractUITagTest {
+
+    private static final String nonceVal = "r4andom";
+
+    public void testScriptTagAttributes() {
+        ScriptTag tag = new ScriptTag();
+        tag.setSrc("mysrc.js");
+        tag.setAsync("false");
+        tag.setType("module");
+        tag.setAsync("false");
+        tag.setNomodule("true");
+        tag.setReferrerpolicy("same-origin");
+        tag.setCrossorigin("anaonymous");
+        String s = writer.toString();
+        assertTrue(s.indexOf("nonce=\"" + nonceVal) > -1);
```

Comment:
we're going to add asserts for each attribute (once we make the change for attributes like async, nomodule in script.ftl)

Reply:
resolved

Reply:
resolved